packages/flutter/lib/src/rendering/object.dart

```diff
@@ -760,6 +860,17 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     markNeedsCompositingBitsUpdate();
   }
 
+  @override
+  void attach(Object newOwner) {
+    super.attach(newOwner);
+    if (owner != null) {
+      if (_needsLayout)
+        owner._nodesNeedingLayout.add(this);
```

Comment:
Is _needsLayout set all the way up to the tree when markNeedsLayout is called?  I would expect that only relayout roots would get added to _nodesNeedingLayout.  Does this add render objects to _nodesNeedingLayout that aren't relayout roots?